pandas/tests/io/test_excel.py

```diff
@@ -1212,6 +1212,15 @@ class and any subclasses, on account of the `autouse=True`
 class TestExcelWriter(_WriterBase):
     # Base class for test cases to run with different Excel writers.
 
+    def test_excel_sheet_size(self):        
+        breaking_shape = (2**20 + 1, 2**14 + 1)
+        arr = np.zeros(shape=breaking_shape)
```

Comment:
Instead of one large `N x M` DataFrame here you could two two DataFrames (N x 1) and (1 x M) where one exceeds the row limit and the other exceeds the column limit. Improves test coverage and should circumvent the MemoryError

Reply:
good thinking!